fhir-search/src/main/java/com/ibm/fhir/search/parameters/IncludesUtil.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.fhir.search.parameters;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.config.FHIRConfigHelper;
+import com.ibm.fhir.core.ResourceType;
+import com.ibm.fhir.model.resource.SearchParameter;
+import com.ibm.fhir.model.type.code.ResourceTypeCode;
+import com.ibm.fhir.model.type.code.SearchParamType.Value;
+import com.ibm.fhir.search.util.SearchHelper;
+
+public class IncludesUtil {
+    private static final Logger LOG = java.util.logging.Logger.getLogger(IncludesUtil.class.getName());
+
+    /**
+     * Computes the list of supported include parameter values that target resource types from the passed resourcesToInclude list.
+     *
+     * @param baseResourceType
+     * @param resourceTypesToInclude
+     * @param searchHelper
+     * @return The list of values to use for the _include search parameter.
+     * @throws Exception
+     */
+    public static List<String> computeIncludesParamValues(String baseResourceType, List<String> resourceTypesToInclude, SearchHelper searchHelper)
+            throws Exception {
+        List<String> result = new ArrayList<>();
+
+        List<String> allowedIncludes = FHIRConfigHelper.getSearchPropertyRestrictions(baseResourceType, FHIRConfigHelper.SEARCH_PROPERTY_TYPE_INCLUDE);
+        Map<ResourceType, List<String>> codesByType = IncludesUtil.getSearchCodesByType(baseResourceType, searchHelper);
+
+        for (Map.Entry<ResourceType, List<String>> entry : codesByType.entrySet()) {
+            if (resourceTypesToInclude.contains(entry.getKey().value())) {
```

Comment:
Multiple calls to List.contains here. Would perform better if resourceTypesToInclude were a Set, which also makes more sense from a semantic point of view too.

Reply:
done